lib/Support/OSCompatPosix.cpp

```diff
@@ -778,10 +778,12 @@ bool unset_env(const char *name) {
 void *SigAltStackLeakSuppressor::stackRoot_{nullptr};
 
 SigAltStackLeakSuppressor::~SigAltStackLeakSuppressor() {
+#if !(defined(TARGET_OS_TV) && TARGET_OS_TV)
```

Comment:
It seems a little safer (and more common) to use something like `#if defined(__APPLE__) && TARGET_OS_TV`. That way, forgetting to include `TargetConditionals.h` will result in a preprocessor error. In practice, this would matter most for code that actually can compile on the target platform, but it seems good to be consistent.